src/topology/category/Top/limits.lean

```diff
@@ -101,3 +102,96 @@ instance forget_preserves_colimits : preserves_colimits (forget : Top.{u} ⥤ Ty
       (colimit_cocone_is_colimit F) (types.colimit_cocone_is_colimit (F ⋙ forget)) } }
 
 end Top
+
+namespace Top
+
+section topological_konig
+
+/-!
+## Topological Kőnig's lemma
+
+A topological version of Kőnig's lemma is that the inverse limit of nonempty compact Hausdorff
+spaces is nonempty.  (Note: this can be generalized further to inverse limits of nonempty compact
+T0 spaces, where all the maps are closed maps; see [Stone1979] --- however there is an erratum
+for Theorem 4 that the element in the inverse limit can have cofinally many components that are
+not closed points.)
+-/
+
+variables {J : Type u} [directed_order J]
+variables (F : Jᵒᵖ ⥤ Top.{u})
+
+/--
+The partial sections of an inverse system of topological spaces from an index `j` are sections
+when restricted to all objects less than or equal to `j`.
+-/
+def partial_sections (j : Jᵒᵖ) : set (Π j, F.obj j) :=
+{ u | ∀ {j'} (f : j ⟶ j'), F.map f (u j) = u j'}
+
+lemma partial_sections.nonempty [Π (j : Jᵒᵖ), nonempty (F.obj j)] (j : Jᵒᵖ) :
+  (partial_sections F j).nonempty :=
+begin
+  haveI := classical.dec_pred (λ (j' : Jᵒᵖ), j'.unop ≤ j.unop),
```

Comment:
```suggestion
  classical,
```
?

Reply:
In some earlier version `classical` was bringing in some conflicting instances so I pulled in the bare minimum, but it works fine now.